oracle_server/oracle_server.go

```diff
@@ -214,42 +221,17 @@ func (os *OracleServer) gcRoundData() {
 }
 
 func (os *OracleServer) handleConnectivityError() {
-	if os.client != nil {
-		os.client.Close()
-		os.client = nil
-	}
-	os.subRoundEvent.Unsubscribe()
-	os.subSymbolsEvent.Unsubscribe()
+	os.lostSync = true
 }
 
 func (os *OracleServer) checkHealth() {
-	// if the web socket was drops my remote peer, the client will be reset into nil.
-	if os.client == nil {
-		// rebuild the connection with autonity L1 node.
-		// connect to autonity node via web socket
-		var err error
-		os.client, err = os.dialer.Dial(os.l1WSUrl)
-		if err != nil {
-			os.logger.Error("dail L1 node", "error", err.Error())
-			return
-		}
-
-		// bind client with oracle contract address
-		os.logger.Info("binding with oracle contract", "address", types.OracleContractAddress.String())
-		os.oracleContract, err = contract.NewOracle(types.OracleContractAddress, os.client)
-		if err != nil {
-			os.logger.Error("binding oracle contract", "error", err.Error())
-			return
-		}
-
-		err = os.syncStates()
+	if os.lostSync {
+		err := os.syncStates()
```

Comment:
How does this work? in case we lose connection we keep trying to sync? But are we re-dialing or using the same connection?